Find a functionally equivalent code:com.android.externalstorage.ExternalStorageProvider.removeExtension:COMMENT
Method Modifier: private     static      
Comment:/**
 * Remove file extension from name, but only if exact MIME type mapping
 * exists. This means we can reapply the extension later.
 */

@@ -1,7 +1,7 @@
 {
     final int lastDot = name.lastIndexOf('.');
     if (lastDot >= 0) {
-        final String extension = name.substring(lastDot + 1);
+        final String extension = name.substring(lastDot + 1).toLowerCase();
         final String nameMime = MimeTypeMap.getSingleton().getMimeTypeFromExtension(extension);
         if (mimeType.equals(nameMime)) {
             return name.substring(0, lastDot);

